### PR TITLE
3765 Enhance `create_multigpu_supervised_XXX` for distributed

### LINKED ISSUE #3765: `create_multigpu_supervised_trainer` can't work with multiple devices
**Is your feature request related to a problem? Please describe.**
Got feedback that when `distributed=True` in `create_multigpu_supervised_trainer`, the doc-string and typehints show it supports multiple devices and output the first device:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/engines/multi_gpu_supervised_trainer.py#L77
But actually, it can't work with multiple devices and the `None` is against PyTorch API:
```
trainer = create_multigpu_supervised_trainer(net, opt, fake_loss, None, distributed=True)
  File "/workspace/data/medical/MONAI/monai/engines/multi_gpu_supervised_trainer.py", line 90, in create_multigpu_supervised_trainer
    net = DistributedDataParallel(net, device_ids=devices_)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 518, in __init__
    self._log_and_throw(
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 642, in _log_and_throw
    raise err_type(err_msg)
ValueError: device_ids can only be None or contain a single element.
```
We don't have unit test to cover `distributed=True` so far.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3765 .

### Description
This PR enhanced `create_multigpu_supervised_trainer` for devices when `distributed=True`, and added distributed tests.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_parallel_execution_dist.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class DistributedTestParallelExecution(DistTestCase):
+    @DistCall(nnodes=1, nproc_per_node=2)
+    def test_distributed(self):
+        device = torch.device(f"cuda:{dist.get_rank()}")
+        net = torch.nn.Conv2d(1, 1, 3, padding=1).to(device)
+        opt = torch.optim.Adam(net.parameters(), 1e-3)
+
+        trainer = create_multigpu_supervised_trainer(net, opt, fake_loss, [device], distributed=True)
+
+        trainer.run(fake_data_stream(), 2, 2)
```
Comment (wyli) [inline tests/test_parallel_execution_dist.py:39]: should this one check the engine results?

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

File: tests/test_parallel_execution_dist.py (hunk 1 above)
Comment (Nic-Ma) [reply tests/test_parallel_execution_dist.py:39]: Sure, added in the latest commit.

Thanks.

**Pull request merged**